webpack/components/NotificationMenu.js

```diff
@@ -0,0 +1,41 @@
+import React, { Component, PropTypes } from 'react';
+import Notification from './Notification';
+import { readNotification } from '../actions/notification';
+
+export default class NotificationMenu extends Component {
+  constructor(props){
+    super(props);
+
+    this.state = {
+      notifications: props.notifications
+    };
+  }
+
+  notificationClick(id, url) {
+    readNotification(id);
```

Comment:
This invokes the readNotification function (duh). This function returns a flux/redux action, which we ignore. When the action is created in readNotification, the payload is set to a promise, which is the axios.post. That promise will resolve itself, which is why I think the read notifications make it to the server. However, this action is not dispatched in any way, and it will never make it to a redux store, which isn't configured on this page. For now, I would recommend refactoring readNotification to be a plain call to axios. We can make this code redux aware later.